pkg/limiter/fixed_test.go

```diff
@@ -1,26 +1,46 @@
 package limiter_test
 
 import (
+	"context"
 	"testing"
+	"time"
 
 	"github.com/influxdata/influxdb/v2/pkg/limiter"
+	"github.com/stretchr/testify/require"
 )
 
 func TestFixed_Available(t *testing.T) {
 	f := limiter.NewFixed(10)
-	if exp, got := 10, f.Available(); exp != got {
-		t.Fatalf("available mismatch: exp %v, got %v", exp, got)
-	}
+	require.Equal(t, 10, f.Available())
 
-	f.Take()
-
-	if exp, got := 9, f.Available(); exp != got {
-		t.Fatalf("available mismatch: exp %v, got %v", exp, got)
-	}
+	require.NoError(t, f.Take(context.Background()))
+	require.Equal(t, 9, f.Available())
 
 	f.Release()
+	require.Equal(t, 10, f.Available())
+}
+
+func TestFixed_Timeout(t *testing.T) {
+	f := limiter.NewFixed(1)
+	require.NoError(t, f.Take(context.Background()))
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+	err := f.Take(ctx)
+	require.Error(t, err)
+	require.Equal(t, "context deadline exceeded", err.Error())
+}
+
+func TestFixed_Canceled(t *testing.T) {
+	f := limiter.NewFixed(1)
+	require.NoError(t, f.Take(context.Background()))
 
-	if exp, got := 10, f.Available(); exp != got {
-		t.Fatalf("available mismatch: exp %v, got %v", exp, got)
-	}
+	ctx, cancel := context.WithTimeout(context.Background(), 30 * time.Second)
+	go func() {
+		time.Sleep(time.Second)
```

Comment:
Similar question here...might be able to use a smaller time duration rather than a second to keep the tests running more quickly, even though it probably doesn't make any difference in the grand scheme of things...unless there is a specific reason to use 1 second.